### PR TITLE
Expose pipecd ops metrics

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cli/cmd.go [hunk 1]
```diff
@@ -142,15 +145,41 @@ type MetricsBuilder interface {
 	Build() (io.Reader, error)
 }
 
-func (t Telemetry) CustomMetricsHandlerFor(mb MetricsBuilder) http.Handler {
+func (t Telemetry) CustomMetricsHandlerFor(reg *prometheus.Registry, mb MetricsBuilder) http.Handler {
 	if t.Flags.Metrics {
```
Comment (nghialv) [inline pkg/cli/cmd.go:149]: nit: 
``` golang
if !t.Flags.Metrics {
        var empty http.HandlerFunc = func(w http.ResponseWriter, r *http.Request) {
		w.Write([]byte(""))
	}
	return empty
}

File: pkg/cli/cmd.go [hunk 2]
```diff
@@ -142,15 +145,41 @@ type MetricsBuilder interface {
 	Build() (io.Reader, error)
 }
 
-func (t Telemetry) CustomMetricsHandlerFor(mb MetricsBuilder) http.Handler {
+func (t Telemetry) CustomMetricsHandlerFor(reg *prometheus.Registry, mb MetricsBuilder) http.Handler {
```
Comment (nghialv) [inline pkg/cli/cmd.go:148]: Passing `Gatherer` instead of `Registry`.

File: pkg/cli/cmd.go (hunk 1 above)
Comment (khanhtc1202) [reply pkg/cli/cmd.go:149]: I get your point. 👍  Lets keep it as is to unify with others for now, will address it by another PR (apply this pattern to others metrics handler as well)

Comment (khanhtc1202) [reply pkg/cli/cmd.go:149]: Changed my mind, will address it in this PR and apply only for this handler, lol

File: pkg/cli/cmd.go (hunk 2 above)
Comment (khanhtc1202) [reply pkg/cli/cmd.go]: Nice catch 👍 Address by https://github.com/pipe-cd/pipe/pull/2279/commits/6513a2d87a265e2d656c8514a66d3ee2d8782886 🙏

File: pkg/cli/cmd.go (hunk 1 above)
Comment (khanhtc1202) [reply pkg/cli/cmd.go]: Addressed by https://github.com/pipe-cd/pipe/pull/2279/commits/6513a2d87a265e2d656c8514a66d3ee2d8782886 🙏

**Comment by nghialv** [comment]
Looks good!
/lgtm

**Comment by nakabonne** [comment]
Yup, looks perfect ;)
/approve

**Pull request merged**